src/argus_htmx/themes/views.py

```diff
@@ -21,33 +21,43 @@ class ThemeListView(ListView):
 
     def setup(self, request, *args, **kwargs):
         super().setup(request, *args, **kwargs)
-        self.themes = sorted(get_theme_names())
+        self.themes = THEME_NAMES
 
     def get_queryset(self):
         return self.themes
 
     def post(self, request, *args, **kwargs):
-        theme = request.POST.get("theme")
-        if theme in self.themes:
-            request.session["theme"] = theme
+        prefs = request.user.get_preferences("argus_htmx")
+        form = prefs.FORMS["theme"](request.POST)
+        if form.is_valid():
+            theme = form.cleaned_data["theme"]
+            prefs.save_preference("theme", theme)
             messages.success(request, f'Switched theme to "{theme}"')
         else:
-            messages.warning(request, f'Failed to switch to theme "{theme}", no such theme installed')
+            messages.warning(request, "Did not switch theme, theme not installed")
         return HttpResponseRedirect("")
 
 
 @require_GET
 def theme_names(request: HtmxHttpRequest) -> HttpResponse:
-    themes = sorted(get_theme_names())
+    themes = THEME_NAMES
     return render(request, "htmx/themes/theme_list.html", {"theme_list": themes})
 
 
 @require_POST
 def change_theme(request: HtmxHttpRequest) -> HttpResponse:
-    themes = get_theme_names()
-    theme = request.POST.get("theme")
-    if theme in themes:
-        request.session["theme"] = theme
-        messages.success(request, f'Switched theme to "{theme}"')
-        return HttpResponse(f"{theme}")
+    prefs = request.user.get_preferences("argus_htmx")
```

Comment:
What about unauthenticated users?

Reply:
We'd probably have to copy preferences to the session. The session (depending on how it is stored) has a max. size though.

Reply:
The session only needs to store preferences that are different from the default, so that's probably ok. Also, I'm not quite sure how big the max size is?

Reply:
Session storage in a cookie is limited by max cookie size. Sessions can also be stored in for instance the database, with just the key in a cookie, and then there are no limits.

Reply:
As far as I can see, sessions are stored in db by default (using the contrib.sessions app)

Reply:
Right now it is stored in session iff the user is not authenticated.

Reply:
what setting governs that?